### PR TITLE
Fix for retaining the main language when closing splitview (Missing culture in mapping options)

### PR DESCRIPTION
**Pull request by vwa-software**
Related to issues:

[https://github.com/umbraco/Umbraco-CMS/issues/8335](https://github.com/umbraco/Umbraco-CMS/issues/8335)

[https://github.com/umbraco/Umbraco-CMS/issues/11120](https://github.com/umbraco/Umbraco-CMS/issues/11120)
### Description
When closing the splitview, the main culture is sometimes lost, resulting in server exception.

### Steps to reproduce:

* Make sure your site has 2 languages
* Create 2 container content nodes, add  1 or more child items to each.
Schermafbeelding1
* Open 1 child item to edit
* Open splitview for the second culture
* Close one of the splitviews
Schermafbeelding2
* now the mculture is lost in the rout!!!!
Schermafbeelding3
* Navigate to the second container item.
Schermafbeelding4
*  This throws an exception on the server: Missing culture in mapping options
Schermafbeelding5

After this fix the main culture remains in the route.

### CODE REVIEW

File: src/Umbraco.Web.UI.Client/src/common/directives/components/content/umbvariantcontenteditors.directive.js [hunk 1]
```diff
@@ -197,7 +197,11 @@
             
             //update the current culture to reflect the last open variant (closing the split view corresponds to selecting the other variant)
             const culture = vm.editors[0].content.language ? vm.editors[0].content.language.culture : null;
-
+            
+            //retain the main culture
+            const mculture = $location.search().mculture;
+            $location.search({ "cculture": culture, 'mculture': mculture, "csegment": vm.editors[0].content.segment});
+          
             $location.search({"cculture": culture, "csegment": vm.editors[0].content.segment});
```
Comment (mikecp) [inline src/Umbraco.Web.UI.Client/src/common/directives/components/content/umbvariantcontenteditors.directive.js:205]: Hello @vwa-software,
I was wondering if there was a specific reason why there are 2 `$location.search`  performed one after the other, or could the "old one" on line 205 be removed?

File: src/Umbraco.Web.UI.Client/src/common/directives/components/content/umbvariantcontenteditors.directive.js [hunk 2]
```diff
@@ -197,7 +197,11 @@
             
             //update the current culture to reflect the last open variant (closing the split view corresponds to selecting the other variant)
             const culture = vm.editors[0].content.language ? vm.editors[0].content.language.culture : null;
-
+            
+            //retain the main culture
+            const mculture = $location.search().mculture;
+            $location.search({ "cculture": culture, 'mculture': mculture, "csegment": vm.editors[0].content.segment});
+
```
Comment (mikecp) [inline src/Umbraco.Web.UI.Client/src/common/directives/components/content/umbvariantcontenteditors.directive.js:204]: Hello @vwa-software,
I was wondering if there was a specific reason why there are 2 $location.search performed one after the other, or could the "old one" on line 205 be removed?

Comment (vwa-software) [reply src/Umbraco.Web.UI.Client/src/common/directives/components/content/umbvariantcontenteditors.directive.js:204]: That's my mistake, yes line 205 can be removed.

**Comment by nul800sebastiaan** [comment]
Thanks for the help @vwa-software! The fix looks fine but I just can't seem to get into the scenario that you illustrate, so before even updating to this newest code, I've looked in Umbraco 9.2 and in 9.3 and I just can't make it fail and produce the "Missing culture in mapping options" error.

I've tried to configure the listview to show the content first or not, to see if that makes a difference:
image


But I can't get to an error state.. what am I doing wrong? 😅 


https://user-images.githubusercontent.com/304656/155877426-93e19a3f-e2ef-4bd4-96a4-9af0899e0e9b.mp4

**Comment by nul800sebastiaan** [comment]
Oh I was missing subitems, have tried those now as well.. still don't have a problem, so weird! 

https://user-images.githubusercontent.com/304656/155877622-d5fa13c9-90d1-4187-9f00-b9d9510a47bc.mp4

**Comment by mikecp** [comment]
Hi @vwa-software , @nul800sebastiaan ,
Same here, I can't reproduce the issue. But I do think I handled another PR a while ago that was handling this mculture issue on a more general basis (but I can't find it back 😢).  But it could be that this specific issue was solved by that other one...

**Comment by nul800sebastiaan** [comment]
Yeah, I saw some related PRs, not necessarily solving this exact problem, but I am guessing they accidentally did solve the problem! 

@vwa-software I would ask you to try again in the latest version of v9 and see if you can still reproduce, https://www.screentogif.com/ is a really excellent free screen recorder, which also produces MP4s! Maybe I am missing something important.

For now I'll close this PR, if we can reproduce, we can have another look at it! 👍 Thanks!

**Pull request closed without merging**